src/Nano_transcriber.cpp

```diff
@@ -0,0 +1,168 @@
+#include <sensor_msgs/Range.h>
+#include <iarc7_msgs/Nano.h>
+#include <iarc7_msgs/FlowVector.h>
+#include <iarc7_msgs/Float64Stamped.h>
+#include <iarc7_msgs/PlanarThrottle.h>
+#include <iarc7_msgs/ESCCommand.h>
+#include <iarc7_msgs/OrientationThrottleStamped.h>
+#include <iarc7_msgs/FlightControllerStatus.h>
+#include <sensor_msgs/Joy.h>
+#include <ros/ros.h>
+
+void distributeMessages(iarc7_msgs::Nano nano_info, 
+                        ros::Publisher& opticalflow_pub,
+                        ros::Publisher& rangefinder_pub,
+                        ros::Publisher& long_range_pub,
+                        ros::Publisher& battery_publisher){
+
+    sensor_msgs::Range long_range_msg;
+    sensor_msgs::Range short_range_msg;
+    iarc7_msgs::FlowVector flow_msg;
+    iarc7_msgs::Float64Stamped battery_voltage;
+
+    long_range_msg.range = nano_info.long_range;
+    long_range_msg.min_range = 0.5;
+    long_range_msg.max_range = 10;
+    ros::Duration long_range_offset = ros::Duration().fromNSec((int64_t)(nano_info.long_range_offset * 10000));
+    long_range_msg.header.stamp = nano_info.msg_received - long_range_offset;
+    long_range_msg.radiation_type = sensor_msgs::Range::INFRARED;
+    long_range_msg.field_of_view = 0.3;
+    long_range_msg.header.frame_id = "/tfmini";
+    if(nano_info.long_range_offset > 0) {
+        long_range_pub.publish(long_range_msg);
+    }
+
+    short_range_msg.range = nano_info.short_range;
+    short_range_msg.min_range = 0.01;
+    short_range_msg.max_range = 0.800;
+    ros::Duration short_range_offset = ros::Duration().fromNSec((int64_t)(nano_info.short_range_offset * 10000));
+    short_range_msg.header.stamp =  nano_info.msg_received - short_range_offset;
+    short_range_msg.radiation_type = sensor_msgs::Range::INFRARED;
+    short_range_msg.field_of_view = 0.3;
+    short_range_msg.header.frame_id = "/short_distance_lidar";
+    if(nano_info.short_range_offset > 0)
+    {
+        rangefinder_pub.publish(short_range_msg);
+    }
+    
+    ros::Duration flow_offset = ros::Duration().fromNSec((int64_t)(nano_info.short_range_offset * 10000));
+    flow_msg.header.stamp = (nano_info.msg_received - flow_offset);
+    flow_msg.deltaX = nano_info.deltaX;
+    flow_msg.deltaY = nano_info.deltaY;
+    if(nano_info.flow_board_offset > 0)
+    {
+        opticalflow_pub.publish(flow_msg);
+    }
+
+    battery_voltage.data = nano_info.battery_voltage;
+    battery_voltage.header.stamp = nano_info.msg_received;
+    battery_publisher.publish(battery_voltage);
+
+    return;
+}
+
+void sendESCCommand(iarc7_msgs::PlanarThrottle esc_cmnds,
+                    ros::Publisher& esc_cmd_pub,
+                    bool activateSideRotors)
+{
+
+    iarc7_msgs::ESCCommand esc_cmd;
+
+    if(activateSideRotors)
+    {
+        esc_cmd.front_motor_PWM = (esc_cmnds.front_throttle*125) + 125;
```

Comment:
Why the 125?

Reply:
This is to turn a number from 0-1 into a corresponding number in the int8 range, which corresponds to the duty cycle sent to each side rotor.